src/SerializationGraphNavigator.php

```diff
@@ -108,22 +110,22 @@ public function accept($data, array $type = null, Context $context)
                 return $visitor->visitNull($data, $type);
 
             case 'string':
-                return $visitor->visitString($data, $type);
+                return $visitor->visitString((string)$data, $type);
```

Comment:
Are these changes are potentially BC breaking, but not by this PR, by already existing type hints. This only eliminates runtime type juggling.